src/OrbitGl/TrackManager.cpp

```diff
@@ -177,34 +176,52 @@ void TrackManager::SetFilter(const std::string& filter) {
   UpdateVisibleTrackList();
 }
 
+// This function assumes visible track list is updated.
+float TrackManager::GetVisibleTracksTotalHeight() const {
+  // Top and Bottom Margin. TODO: Margins should be treated in a different way (http://b/192070555).
+  float track_height = layout_->GetSchedulerTrackOffset() + layout_->GetBottomMargin();
+
+  // Track height including space between them
+  for (auto& track : visible_tracks_) {
+    track_height += (track->GetHeight() + layout_->GetSpaceBetweenTracks());
+  }
+  return track_height;
+}
+
 void TrackManager::UpdateVisibleTrackList() {
```

Comment:
Nit: Maybe this method should explicitly return right away if `visible_track_list_needs_update_ == false`. This way, it can't accidentally be called when not needed, and it's very clear when reading the method that it's only called when needed.

I was reading this and at first I thought this would call `time_graph_->RequestUpdate()` in every frame.

Reply:
This function is a private function which only should have been called once. You are suggesting to move that if from there to here?

Reply:
You could move the `if`, yes - as I wrote, it's just a nitpick, it just might make it easier for someone reading the trackmanager code when this is happening.

Reply:
Replaced by 2 checks, will upload the commit soon.